guides/creating-a-custom-form-field-control.md

```diff
@@ -159,21 +159,48 @@ private _placeholder: string;
 
 This property allows the form field control to specify the `@angular/forms` control that is bound to
 this component. Since we haven't set up our component to act as a `ControlValueAccessor`, we'll just
-set this to `null` in our component. In any real component, you would probably want to implement
-`ControlValueAccessor` so that your component can work with `formControl` and `ngModel`.
+set this to `null` in our component. 
 
 ```ts
-ngControl = null;
+ngControl: NgControl = null;
 ```
 
-If you did implement `ControlValueAccessor`, you could simply inject the `NgControl` and make it
-publicly available. (For additional information about `ControlValueAccessor` see the
-[API docs](https://angular.io/api/forms/ControlValueAccessor).)
+In any real component, you would probably want to implement `ControlValueAccessor` so that your component 
+can work with `formControl` and `ngModel`. If you do implement `ControlValueAccessor`, you will need to get
+a reference to the `NgControl` associated with yoru control, and make it publicly available. 
+
+One way to do this is to inject it:
 
 ```ts
 constructor(..., @Optional() @Self() public ngControl: NgControl) { ... }
 ```
 
+...however, if your component implements `ControlValueAccessor`, it may already provide `NG_VALUE_ACCESSOR` and attempting to inject `NgControl` this may cause a dependency injection error ("Cannot instantiate cyclic dependency"). 
+
+If necessary, inject the `Injector` itself and use it to get the `NgControl` directly:
+
+```ts
+
+ngControl: NgControl;
+
+constructor(
+    private readonly _injector: Injector,
+    // ...
+) { }
+
+// ... 
+
+ngAfterViewInit() {
+    this.ngControl = this._injector.get(NgControl);
```

Comment:
I don't think that this is the proper way to handle it either. Instead of providing the `NG_VALUE_ACCESSOR`, you can set the accessor the same way we do for `mat-select`. See https://github.com/angular/material2/blob/master/src/lib/select/select.ts#L489

Reply:
Ah, I saw two approaches and picked the other one. Not knowing enough about Angular internals it’s hard to say which is better.  I’ll revise.